### PR TITLE
feat: OpenAPI 3.1 schema -> Flatfile Blueprint plugin

### PR DESCRIPTION
**Pull request by carlbrugger**
**Guide:** https://github.com/FlatFilers/Guides/pull/974

This PR introduces the `@flatfile/plugin-convert-openapi-schema` plugin to configure a Flatfile Space based on a provided OpenAPI 3.1 schema.

## Basic Usage

```js
listener.namespace(
  ['space:OpenAPI-simple'],
  configureSpaceWithOpenAPI({
    workbooks: [
      {
        source:
          'https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v3.1/webhook-example.json',
        sheets: [{ name: 'old pets', model: 'Pet' }],
      },
    ],
    debug: true,
  })
)
```

## Advanced Usage

```js
const workbookActions: Flatfile.Action[] = [
  {
    operation: 'submitActionFg',
    mode: 'foreground',
    label: 'Submit data',
    type: 'string',
    description: 'Submit this data to a webhook.',
    primary: true,
  },
  {
    operation: 'downloadWorkbook',
    mode: 'foreground',
    label: 'Download Workbook',
    description: 'Downloads Excel Workbook of Data',
  },
]
const sheetActions: Flatfile.Action[] = [
  {
    operation: 'duplicateSheet',
    mode: 'foreground',
    label: 'Duplicate',
    description: 'Duplicate this sheet.',
    primary: true,
  },
]
const sheets = [
  {
    model: 'AccountingAttachmentResponse',
    slug: 'accountingAttachmentResponse',
  },
  { model: 'Account', actions: sheetActions },
  {
    model: 'Address',
    slug: 'addresses',
    name: 'Addresses',
    actions: sheetActions,
  },
  { model: 'Attachment', slug: 'attachments' },
  { model: 'BalanceSheet', slug: 'balanceSheets' },
  { model: 'CashFlowStatement', slug: 'cashFlowStatements' },
  { model: 'CompanyInfo', slug: 'companyInfo' },
  { model: 'CreditNote', slug: 'creditNotes' },
  { model: 'Expense', slug: 'expenses' },
  { model: 'IncomeStatement', slug: 'incomeStatements' },
  { model: 'Invoice', slug: 'invoices' },
  { model: 'Item', slug: 'items' },
  { model: 'JournalEntry', slug: 'journalEntries' },
  { model: 'Payment', slug: 'payments' },
  { model: 'PhoneNumber', slug: 'phoneNumbers' },
  { model: 'PurchaseOrder', slug: 'purchaseOrders' },
  { model: 'TaxRate', slug: 'taxRates' },
  { model: 'TrackingCategory', slug: 'trackingCategories' },
  { model: 'Transaction', slug: 'transactions' },
  { model: 'VendorCredit', slug: 'vendorCredits' },
]

const callback = async (event, workbookIds, tick) => {
  const { spaceId } = event.context
  await api.documents.create(spaceId, {
    title: 'Welcome',
    body: `<div>
      <h1>Welcome!</h1>
      <h2>To get started, follow these steps:</h2>
      <h2>1. Step One</h2>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.</p>
      <h3>Remember, if you need any assistance, you can always refer back to this page by clicking "Welcome" in the left-hand sidebar!</h3>
      </div>`,
  })
  await tick(80, 'Document created')
}

listener.namespace(
  ['space:OpenAPI-advanced'],
  configureSpaceWithOpenAPI(
    {
      workbooks: [
        {
          source: 'https://api.merge.dev/api/accounting/v1/schema',
          sheets,
          actions: workbookActions,
        },
      ],
    },
    callback
  )
)
```

### CODE REVIEW

File: plugins/openapi/package.json [hunk 1]
```diff
@@ -3,15 +3,32 @@
   "version": "0.0.2",
```
Comment (ashleygmulligan2) [inline plugins/openapi/package.json:3]: let's rename this to @flatfile/plugin-convert-openapi

**Review by ashleygmulligan2 — CHANGES_REQUESTED** [review]

**Comment by ashleygmulligan2** [comment]
will come back to review this once we can look at the Guide PR at the same time

File: plugins/zip-extractor/src/zip.extractor.e2e.spec.ts [hunk 2]
```diff
@@ -40,6 +40,6 @@ describe('ZipExtractor e2e', () => {
       await listener.waitFor('file:created', 4)
       const filesPostUpload = await getFiles(spaceId)
       expect(filesPostUpload.length).toBe(4)
-    })
+    }, 15_000)
```
Comment (bangarang) [inline plugins/zip-extractor/src/zip.extractor.e2e.spec.ts:43]: 15s is a while! 😭

**Review by bangarang — APPROVED** [review]

File: plugins/zip-extractor/src/zip.extractor.e2e.spec.ts (hunk 2 above)
Comment (carlbrugger) [reply plugins/zip-extractor/src/zip.extractor.e2e.spec.ts:43]: Usually takes the default 5, but sometimes it takes a little longer. Just got tired of the test failing occasionally.

**Pull request merged**